aquia-inc/template-vite-react
# build(deps): update React and related deps by minor versions

## Summary

Upgraded several dependencies to their latest versions to ensure the application remains up-to-date with the latest features, improvements, and security patches.

### Added

- None

### Changed

- Upgraded `@emotion/react` from `^11.11.4` to `^11.13.0`
- Upgraded `@emotion/styled` from `^11.11.5` to `^11.13.0`
- Upgraded `@fortawesome/fontawesome-free` from `^6.5.1` to `^6.6.0`
- Upgraded `@hookform/resolvers` from `^3.3.4` to `^3.9.0`
- Upgraded `@mui/icons-material` from `^5.15.18` to `^5.16.6`
- Upgraded `@mui/lab` from `^5.0.0-alpha.170` to `^5.0.0-alpha.173`
- Upgraded `@mui/material` from `^5.15.18` to `^5.16.6`
- Upgraded `@mui/x-data-grid` from `^6.19.11` to `^6.20.4`
- Upgraded `react` from `^18.2.0` to `^18.3.1`
- Upgraded `react-dom` from `^18.2.0` to `^18.3.1`
- Upgraded `react-hook-form` from `^7.50.1` to `^7.52.2`
- Upgraded `react-router-dom` from `^6.25.0` to `^6.26.0`
- Upgraded `yup` from `^1.3.3` to `^1.4.0`

### Deprecated

- None

### Removed

- None

### Fixed

- None

## How to test

1. Checkout this branch: `git checkout origin/build/dependencies/2-react`
2. Run `yarn` and ensure all dependencies are correctly installed without duplication and that there are no issues related to missing or incompatible dependencies.
3. Run `yarn test` and verify that all tests run and pass.
4. Run `yarn dev` and verify that the application runs as expected.
5. Run `yarn build` and verify that the application builds as expected.
6. Run `yarn build-storybook` and verify the Storybook build succeeds.
